test/swar/BasicOperations.cpp

```diff
@@ -299,6 +307,64 @@ GE_MSB_TEST(0x7777'7777,
             0x0123'4567,
             0x8888'8888)
 
+// Replicate the msb off tests with the lessthan that allows msb on
+#define GE_MSB_ON_TEST(left, right, result) static_assert(result == greaterEqual<4, u32>(SWAR<4, u32>(left), SWAR<4, u32>(right)).value());
+
+GE_MSB_ON_TEST(0x1000'0010,
+               0x0111'1101,
+               0x8000'0080)
+GE_MSB_ON_TEST(0x4333'3343,
+               0x4444'4444,
+               0x8000'0080)
+GE_MSB_ON_TEST(0x0550'0110,
+               0x0110'0550,
+               0x8888'8008)
+GE_MSB_ON_TEST(0x4771'1414,
+               0x4641'1774,
+               0x8888'8008)
+GE_MSB_ON_TEST(0x0123'4567,
+               0x0000'0000,
+               0x8888'8888)
+GE_MSB_ON_TEST(0x0123'4567,
+               0x7777'7777,
+               0x0000'0008)
+GE_MSB_ON_TEST(0x0000'0000,
+               0x0123'4567,
+               0x8000'0000)
+GE_MSB_ON_TEST(0x7777'7777,
+               0x0123'4567,
+               0x8888'8888)
+
+TEST_CASE(
+    "greaterEqualMSBOn",
+    "[swar][signed-swar][unsigned-swar]"
```

Comment:
Wait, does this implementation of greaterEqual is aware of two's complement?
Why does this have the tag `signed-swar`?

Reply:
That's leftover from a previous mis-reading of either that Lamport paper or the Knuth book.  I'm not sure there is a way to that easily (though I'm going to try it at some point, no doubt)